File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/client/renderer/level/EntityRenderer.java
package pize.tests.voxelgame.client.renderer.level;

import pize.app.Disposable;
import pize.tests.voxelgame.client.control.camera.GameCamera;
import pize.tests.voxelgame.client.entity.RemotePlayer;
import pize.tests.voxelgame.client.entity.model.PlayerModel;
import pize.tests.voxelgame.client.level.ClientLevel;
import pize.tests.voxelgame.main.entity.Entity;

public class EntityRenderer implements Disposable{
    
    final LevelRenderer levelRenderer;
    
    public EntityRenderer(LevelRenderer levelRenderer){
        this.levelRenderer = levelRenderer;
    }
    
    
    public void render(GameCamera camera){
        final ClientLevel level = levelRenderer.getGameRenderer().getSession().getGame().getLevel();
        
        for(Entity entity : level.getEntities()){
            
            // Remote players
            if(entity instanceof RemotePlayer remotePlayer){
                remotePlayer.updateInterpolation();
                
                final PlayerModel model = remotePlayer.getModel();
                if(model != null){
                    model.animate();
                    model.render(camera);
                }
            }
            
        }
    }
    
    @Override
    public void dispose(){ }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/client/renderer/level/EntityRenderer.kt
package pize.tests.voxelgame.client.renderer.level

import pize.app.Disposable
import pize.tests.voxelgame.client.control.camera.GameCamera
import pize.tests.voxelgame.client.entity.RemotePlayer
import pize.tests.voxelgame.client.entity.model.PlayerModel

class EntityRenderer(val levelRenderer: LevelRenderer) : Disposable {
    fun render(camera: GameCamera?) {
        val level = levelRenderer.gameRenderer.session.game.level
        for (entity in level.entities) {

            // Remote players
            if (entity is RemotePlayer) {
                entity.updateInterpolation()
                val model: PlayerModel = entity.model
                if (model != null) {
                    model.animate()
                    model.render(camera)
                }
            }
        }
    }

    override fun dispose() {}
}
